### PR TITLE
WIP: Support timeout option when pulling image.

### PR DESCRIPTION
**Pull request by haosdent**

### CODE REVIEW

**Comment by haosdent** [comment]
@Random-Liu could you help to review this?

**Comment by mikebrow** [comment]
thx for the tip commit.. Commits need to be signed.  With your add-pull-timeout branch checked out pls "git -s --amend" to this commit and force push the commit back up here.

File: vendor/github.com/containerd/containerd/pull.go [hunk 1]
```diff
@@ -32,8 +33,16 @@ import (
 
 // Pull downloads the provided content into containerd's content store
 // and returns a platform specific image object
-func (c *Client) Pull(ctx context.Context, ref string, opts ...RemoteOpt) (_ Image, retErr error) {
+func (c *Client) Pull(ctx context.Context, ref string, timeout time.Duration, opts ...RemoteOpt) (_ Image, retErr error) {
```
Comment (mikebrow) [inline vendor/github.com/containerd/containerd/pull.go]: To update vendor files we modify vendor.conf at the root of this repo in your branch.. then run the tool "vndr".  vndr will bring in the necessary vendor/ files then you add a commit for your change to vendor.conf and vendor/

Comment (haosdent) [reply vendor/github.com/containerd/containerd/pull.go]: OK, thx!

**Comment by haosdent** [comment]
OK, I just realize kubernetes support pull image timeout before https://github.com/kubernetes/kubernetes/pull/41274 . Close this and create another pr at https://github.com/kubernetes/kubernetes/pull/87641

**Pull request closed without merging**